cross-compile/macosx/create_dmg

```diff
@@ -23,6 +23,9 @@ set -e
 # The path where the installed sigrok libraries/binaries are located.
 PREFIX=$HOME/sr_macosx
 
+# Build for Apple Silicon (y/n)
+APPLE_SILICON=y
```

Comment:
I tested this PR on macOS 14.1.1 / M1 CPU.

As I stated in description:

> APPLE_SILICON option is figurative, I think most fixes are relevant to Intel Mac running latest macOS, so read it as LATEST_MACOS.

I believe it's relevant for macOS 11 and beyond. But since there's no mention in documentation what versions of macOS currently supported by sigrok-util/PulseView it's hard for me to delineate what versions is APPLE_SILICON=`y` or =`n`. That's why I added this changes an option, and not permanent changes, so you can revert to original behaviour by setting `APPLE_SILICON` to `n`.

Reply:
> I tested this PR on macOS 14.1.1 / M1 CPU.


> I believe it's relevant for macOS 11 and beyond. 

That's a good start, and that kind of info should definitely not just be deep in here in a comment thread but at least at the top in the PR submission, commit message etc. Otherwise in 2-3 months when another macos user comes back with issues / fixes, then nobody knows what this apply to. This kind of information gets lost easily which is exactly why , as you say,
> there's no mention in documentation what versions of macOS currently supported by sigrok-util/PulseView